pkg/api/mlflow/dao/convertors/experiment.go

```diff
@@ -33,7 +39,31 @@ func ConvertCreateExperimentToDBModel(req *request.CreateExperimentRequest) *mod
 		}
 	}
 
-	return &experiment
+	if req.ArtifactLocation != "" {
+		u, err := url.Parse(req.ArtifactLocation)
+		if err != nil {
+			return nil, eris.Wrap(err, "error parsing artifact location")
+		}
+
+		switch u.Scheme {
+		case "s3":
+			artifactLocation := fmt.Sprintf("%s://%s", u.Scheme, strings.Trim(u.Host, "/"))
+			if path := strings.Trim(u.Path, "/"); path != "" {
+				artifactLocation = fmt.Sprintf("%s/%s", artifactLocation, path)
+			}
+			experiment.ArtifactLocation = artifactLocation
```

Comment:
why are we doing all of this trimming of slashes? if it's to be able to append experiment IDs and run IDs later, I would argue for using `url.JoinPath` instead

Reply:
this is because of different cases which I've tested:
- `s3://location/`
- s3://location/next`
- s3://location/next/
I can't see any reason to use `url.JoinPath` here.

Reply:
so we have to correctly trim latest slash. can't just use `url.JoinPath` and be happy. in that case the full url will be incorrect.

Reply:
probably I can use `url.JoinPath` but in any case I have to `trim` extra `slashes` that we don't need to be in path.

Reply:
I mean `url.JoinPath` when you append the experiment IDs and run IDs later — but also why are you trimming the `Host` here? and why not just trim the slashes from the path and use `u.String()`?